libraries/Microsoft.Bot.Connector/Authentication/BuiltinBotFrameworkAuthentication.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    internal abstract class BuiltinBotFrameworkAuthentication : BotFrameworkAuthentication
+    {
+        private readonly string _toChannelFromBotOAuthScope;
+        private readonly string _loginEndpoint;
+        private readonly string _callerId;
+        private readonly string _channelService;
+        private readonly ServiceClientCredentialsFactory _credentialFactory;
+        private readonly AuthenticationConfiguration _authConfiguration;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
+
+        protected BuiltinBotFrameworkAuthentication(string toChannelFromBotOAuthScope, string loginEndpoint, string callerId, string channelService, ServiceClientCredentialsFactory credentialFactory, AuthenticationConfiguration authConfiguration, HttpClient httpClient, ILogger logger)
+        {
+            _toChannelFromBotOAuthScope = toChannelFromBotOAuthScope;
```

Comment:
On some it might give an improved error reporting, but they are there because this code delegates to the existing implementation. This class might not actually ship. Ideally we just ship the parameterized version of the story. As it is this this a consistency check with the parameterized class. i.e. if the behavior of the parameterized version is in any way different than the behavior of this code than we have a regression.

Having said that, this code could ship, its correct and internal. It's really the existing code, repackaged a little.

Reply:
Anyhow, do we need null check? No. Well not yet anyhow.